<?xml version="1.0" encoding="UTF-8"?>

<launch>
    
    <arg name="world" default="$(find project)/world/home.world"/>

    <arg name="gui" default="true"/> <!-- Starts Gazebo Interface -->
    <arg name="debug" default="false"/> <!-- Starts Gazebo Debug -->

    <arg name="paused" default="true"/>
    
    <arg name="use_sim_time" default="true" />
    <arg name="headless" default="false" />

    <arg name="gzpose" default="-x 0.0 -y 0.0 -z 0.0 -R 0.0 -P 0.0 -Y 0.0"/>
    <arg name="model" default="$(find project)/dogbot_description/urdf/dogbot.xacro"/>
    
    <arg name="rvizconfig" default="$(find project)/dogbot_control/rviz/urdf.rviz" />

    <arg name="namespace" default="dogbot" />
    <env name="GAZEBO_MODEL_PATH" value="$(find project)/models/walls:$(find project)/models/obstacles:$(find project)/models/robot:$(find project)/models/QR:$(optenv GAZEBO_MODEL_PATH)" />



    <!-- Start World -->
    <include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name" value="$(arg world)"/>
        <arg name="gui" value="$(arg gui)"/>
        <arg name="debug" value="$(arg debug)"/>
        <arg name="paused" value="$(arg paused)"/>
        <arg name="use_sim_time" value="$(arg use_sim_time)" />
        <arg name="headless" value="$headless" />
    </include>

    <!-- Publish Odometry -->
    <node name="odometry_node" pkg="project" type="odometry_node" output="screen"/>

    <param name="robot_description" command="$(find xacro)/xacro $(find project)/dogbot_description/urdf/dogbot.xacro"/>


    <include file="$(find project)/launch/dog.launch">
      <arg name="model" value="$(arg model)" />
      <arg name="paused" value="$(arg paused)"/>
      <arg name="rvizconfig" value="$(arg rvizconfig)" />
    </include>


    <include file="$(find project)/launch/gmapping.launch"/>

    <!-- ROS Control-->
    <rosparam file="$(find project)/conf/control.yaml" command="load" />


    <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false"  output="screen" ns="/dogbot" 
      args="joint_state_controller joint_effort_controllers" />

    
    <node name="service_server" pkg="project" type="service_server"  respawn="false" output="screen" /> 
    <node name="apf" pkg="project" type="apf"  respawn="false" output="screen" /> 
    <node name="opencv_node" pkg="project" type="opencv_node"  respawn="false" output="screen" /> 
    <node name="map_saver" pkg="project" type="map_saver"  respawn="false" output="screen" /> 
    

</launch>
